### ISSUE TITLE
Native ocr lib crash

### ISSUE THREAD

**Issue by GEllickson** [googlesamples/android-vision#269]
Our app uses text and barcode detection, and a very small minority have been experiencing this crash. I can only see the crash through Google Play and not through our Crashlytics integration. The devices have only been various new-ish Samsung devices running Nougat (both 7 and 7.1), but majority with those same specs have been fine. The crash happens consistently with those users though, and effectively renders our in-app camera non-functional for them. 

Is there anything I could do in-app to prevent this or is this a library bug? We do check if the detectors are operational before using them.

```
native: pc 000000000006fc2c  /system/lib64/libc.so (tgkill+8)
  native: pc 000000000006cf40  /system/lib64/libc.so (pthread_kill+64)
  native: pc 00000000000251f8  /system/lib64/libc.so (raise+24)
  native: pc 000000000001ccc4  /system/lib64/libc.so (abort+52)
  native: pc 00000000004750d0  /data/data/com.google.android.gms/app_vision/ocr/libs/arm64-v8a/libocr.so
  native: pc 0000000000475770  /data/data/com.google.android.gms/app_vision/ocr/libs/arm64-v8a/libocr.so
  native: pc 0000000000475814  /data/data/com.google.android.gms/app_vision/ocr/libs/arm64-v8a/libocr.so
  native: pc 0000000000474830  /data/data/com.google.android.gms/app_vision/ocr/libs/arm64-v8a/libocr.so
  native: pc 000000000026cee4  /data/data/com.google.android.gms/app_vision/ocr/libs/arm64-v8a/libocr.so
  native: pc 0000000000271190  /data/data/com.google.android.gms/app_vision/ocr/libs/arm64-v8a/libocr.so
  native: pc 0000000000065794  /data/data/com.google.android.gms/app_vision/ocr/libs/arm64-v8a/libocr.so
  native: pc 0000000000065488  /data/data/com.google.android.gms/app_vision/ocr/libs/arm64-v8a/libocr.so
  native: pc 000000000027caf4  /data/data/com.google.android.gms/app_vision/ocr/libs/arm64-v8a/libocr.so
  native: pc 0000000000280274  /data/data/com.google.android.gms/app_vision/ocr/libs/arm64-v8a/libocr.so
  native: pc 0000000000067c84  /data/data/com.google.android.gms/app_vision/ocr/libs/arm64-v8a/libocr.so
  native: pc 0000000000067f64  /data/data/com.google.android.gms/app_vision/ocr/libs/arm64-v8a/libocr.so
  native: pc 0000000000063dec  /data/data/com.google.android.gms/app_vision/ocr/libs/arm64-v8a/libocr.so
  native: pc 000000000038c5e0  /data/user_de/0/com.google.android.gms/app_chimera/m/0000002b/oat/arm64/DynamiteModulesA_GmsCore_prodmnc_xxhdpi_release.odex

```

**Comment by GEllickson** [comment]
Seeing the crash with a growing number of users, but still all Samsung running 7 or 7.1

Dug deeper into the Google Play console to get a bit more info on the crash. Here's an example:

*** *** *** *** *** *** *** *** *** *** *** *** *** *** *** ***
Build fingerprint: 'Verizon/nobleltevzw/nobleltevzw:7.0/NRD90M/N920VVRS3CQI2:user/release-keys'
Revision: '9'
ABI: 'arm64'
pid: 8702, tid: 8702, name: cout.receipthog  >>> com.infoscout.receipthog <<<
signal 6 (SIGABRT), code -6 (SI_TKILL), fault addr --------
**Abort message: 'timer.cc:38 Check failed: x >= 1 (0 vs. 1)** 
'
    x0   0000000000000000  x1   00000000000021fe  x2   0000000000000006  x3   0000000000000008
    x4   00000000000001eb  x5   0000008000000000  x6   00000074f216a000  x7   0000000000000000
    x8   0000000000000083  x9   ffffffffffffffdf  x10  0000000000000000  x11  0000000000000001
    x12  ffffffffffffffff  x13  0000000000000000  x14  0000000000000000  x15  002d0d2a144eaf41
    x16  00000074f1a29ee0  x17  00000074f19d35ac  x18  00000074ca5ebe6d  x19  00000074f221cb40
    x20  0000000000000006  x21  00000074f221ca98  x22  0000000000000002  x23  00000074cac58480
    x24  00000074b8234780  x25  00000074ce9f27e0  x26  00000074b7b59478  x27  0000007ff091a4e8
    x28  0000000000000000  x29  0000007ff0919670  x30  00000074f19d0a54
    sp   0000007ff0919650  pc   00000074f19d35b4  pstate 0000000060000000

backtrace:
    #00 pc 000000000006b5b4  /system/lib64/libc.so (tgkill+8)
    #01 pc 0000000000068a50  /system/lib64/libc.so (pthread_kill+64)
    #02 pc 0000000000023f68  /system/lib64/libc.so (raise+24)
    #03 pc 000000000001c9ec  /system/lib64/libc.so (abort+52)
    #04 pc 00000000004750d0  /data/data/com.google.android.gms/app_vision/ocr/libs/arm64-v8a/libocr.so
    #05 pc 0000000000475770  /data/data/com.google.android.gms/app_vision/ocr/libs/arm64-v8a/libocr.so
    #06 pc 0000000000475814  /data/data/com.google.android.gms/app_vision/ocr/libs/arm64-v8a/libocr.so
    #07 pc 0000000000474830  /data/data/com.google.android.gms/app_vision/ocr/libs/arm64-v8a/libocr.so
    #08 pc 000000000026cee4  /data/data/com.google.android.gms/app_vision/ocr/libs/arm64-v8a/libocr.so
    #09 pc 0000000000271190  /data/data/com.google.android.gms/app_vision/ocr/libs/arm64-v8a/libocr.so
    #10 pc 0000000000065794  /data/data/com.google.android.gms/app_vision/ocr/libs/arm64-v8a/libocr.so
    #11 pc 0000000000065488  /data/data/com.google.android.gms/app_vision/ocr/libs/arm64-v8a/libocr.so
    #12 pc 000000000027caf4  /data/data/com.google.android.gms/app_vision/ocr/libs/arm64-v8a/libocr.so
    #13 pc 0000000000280274  /data/data/com.google.android.gms/app_vision/ocr/libs/arm64-v8a/libocr.so
    #14 pc 0000000000067c84  /data/data/com.google.android.gms/app_vision/ocr/libs/arm64-v8a/libocr.so
    #15 pc 0000000000067f64  /data/data/com.google.android.gms/app_vision/ocr/libs/arm64-v8a/libocr.so
    #16 pc 0000000000063dec  /data/data/com.google.android.gms/app_vision/ocr/libs/arm64-v8a/libocr.so
    #17 pc 0000000000388250  /data/user_de/0/com.google.android.gms/app_chimera/m/00000054/oat/arm64/DynamiteModulesA_GmsCore_prodmnc_xxhdpi_release.odex (offset 0x4e7000)

**Comment by LucianIacob** [comment]
I found the same crash on my Samsung S6 Edge device, when trying to create a **TextRecognizer**.
The log cat looks like this:
   
    `10-24 10:18:02.401 11605-11605/com.google.android.gms.samples.vision.ocrreader E/native: jni_helper.cc:170 GetContents failed: /data/user/0/com.google.android.gms/app_vision/ocr/data/models/semanticlift_engine_0.2.bincfg
    10-24 10:18:02.403 11605-11605/com.google.android.gms.samples.vision.ocrreader A/native: timer.cc:38 Check failed: x >= 1 (0 vs. 1) 
    10-24 10:18:02.403 11605-11605/com.google.android.gms.samples.vision.ocrreader A/native: terminating.
      --------- beginning of crash
    10-24 10:18:02.404 11605-11605/com.google.android.gms.samples.vision.ocrreader A/libc: Fatal signal 6 (SIGABRT), code -6 in tid 11605 (ision.ocrreader)
                                                                                       
                                                                                       [ 10-24 10:18:02.405  2991: 2991 W/         ]
                                                                                       debuggerd: handling request: pid=11605 uid=11097 gid=11097 tid=11605
    10-24 10:18:02.416 5323-5323/? E/audit: type=1400 audit(1508829482.408:30100): avc:  denied  { search } for  pid=13003 comm="debuggerd64" name="com.google.android.gms" dev="dm-1" ino=524343 scontext=u:r:debuggerd:s0 tcontext=u:object_r:app_data_file:s0:c512,c768 tclass=dir permissive=0 SEPF_SECMOBILE_7.0_0009
    10-24 10:18:02.416 5323-5323/? E/audit: type=1400 audit(1508829482.408:30101): avc:  denied  { search } for  pid=13003 comm="debuggerd64" name="com.google.android.gms" dev="dm-1" ino=459060 scontext=u:r:debuggerd:s0 tcontext=u:object_r:app_data_file:s0:c512,c768 tclass=dir permissive=0 SEPF_SECMOBILE_7.0_0009
    10-24 10:18:02.488 13003-13003/? A/DEBUG:     x0   0000000000000000  x1   0000000000002d55  x2   0000000000000006  x3   0000000000000008
    10-24 10:18:02.488 13003-13003/? A/DEBUG:     x4   00000000000001f6  x5   0000008000000000  x6   00000072dfb9c000  x7   0000000000000000
    10-24 10:18:02.488 13003-13003/? A/DEBUG:     x8   0000000000000083  x9   ffffffffffffffdf  x10  0000000000000000  x11  0000000000000001
    10-24 10:18:02.488 13003-13003/? A/DEBUG:     x12  ffffffffffffffff  x13  0000000000000000  x14  0000000000000000  x15  0018136540e86b52
    10-24 10:18:02.488 13003-13003/? A/DEBUG:     x16  00000072df44fee0  x17  00000072df3f95ac  x18  00000072b8df628d  x19  00000072dfc4eb40
    10-24 10:18:02.488 13003-13003/? A/DEBUG:     x20  0000000000000006  x21  00000072dfc4ea98  x22  0000000000000002  x23  00000072db26d5d0
    10-24 10:18:02.488 13003-13003/? A/DEBUG:     x24  000000729dc7a780  x25  00000072bb7c6510  x26  000000729d59f478  x27  0000007fe72f1da8
    10-24 10:18:02.488 13003-13003/? A/DEBUG:     x28  0000000000000000  x29  0000007fe72f0f30  x30  00000072df3f6a54
    10-24 10:18:02.488 13003-13003/? A/DEBUG:     sp   0000007fe72f0f10  pc   00000072df3f95b4  pstate 0000000060000000
    10-24 10:18:02.490 5323-5323/? E/audit: type=1400 audit(1508829482.478:30102): avc:  denied  { search } for  pid=13003 comm="debuggerd64" name="com.google.android.gms" dev="dm-1" ino=524343 scontext=u:r:debuggerd:s0 tcontext=u:object_r:app_data_file:s0:c512,c768 tclass=dir permissive=0 SEPF_SECMOBILE_7.0_0009
    10-24 10:18:02.498 5323-5323/? E/audit: type=1400 audit(1508829482.488:30103): avc:  denied  { search } for  pid=13003 comm="debuggerd64" name="com.google.android.gms" dev="dm-1" ino=459060 scontext=u:r:debuggerd:s0 tcontext=u:object_r:app_data_file:s0:c512,c768 tclass=dir permissive=0 SEPF_SECMOBILE_7.0_0009
    10-24 10:18:02.498 13003-13003/? A/DEBUG: backtrace:
    10-24 10:18:02.498 13003-13003/? A/DEBUG:     #00 pc 000000000006b5b4  /system/lib64/libc.so (tgkill+8)
    10-24 10:18:02.498 13003-13003/? A/DEBUG:     #01 pc 0000000000068a50  /system/lib64/libc.so (pthread_kill+64)
    10-24 10:18:02.498 13003-13003/? A/DEBUG:     #02 pc 0000000000023f68  /system/lib64/libc.so (raise+24)
    10-24 10:18:02.498 13003-13003/? A/DEBUG:     #03 pc 000000000001c9ec  /system/lib64/libc.so (abort+52)
    10-24 10:18:02.498 13003-13003/? A/DEBUG:     #04 pc 00000000004750d0  /data/data/com.google.android.gms/app_vision/ocr/libs/arm64-v8a/libocr.so
    10-24 10:18:02.499 13003-13003/? A/DEBUG:     #05 pc 0000000000475770  /data/data/com.google.android.gms/app_vision/ocr/libs/arm64-v8a/libocr.so
    10-24 10:18:02.499 13003-13003/? A/DEBUG:     #06 pc 0000000000475814  /data/data/com.google.android.gms/app_vision/ocr/libs/arm64-v8a/libocr.so
    10-24 10:18:02.499 13003-13003/? A/DEBUG:     #07 pc 0000000000474830  /data/data/com.google.android.gms/app_vision/ocr/libs/arm64-v8a/libocr.so
    10-24 10:18:02.499 13003-13003/? A/DEBUG:     #08 pc 000000000026cee4  /data/data/com.google.android.gms/app_vision/ocr/libs/arm64-v8a/libocr.so
    10-24 10:18:02.499 13003-13003/? A/DEBUG:     #09 pc 0000000000271190  /data/data/com.google.android.gms/app_vision/ocr/libs/arm64-v8a/libocr.so
    10-24 10:18:02.499 13003-13003/? A/DEBUG:     #10 pc 0000000000065794  /data/data/com.google.android.gms/app_vision/ocr/libs/arm64-v8a/libocr.so
    10-24 10:18:02.499 13003-13003/? A/DEBUG:     #11 pc 0000000000065488  /data/data/com.google.android.gms/app_vision/ocr/libs/arm64-v8a/libocr.so
    10-24 10:18:02.499 13003-13003/? A/DEBUG:     #12 pc 000000000027caf4  /data/data/com.google.android.gms/app_vision/ocr/libs/arm64-v8a/libocr.so
    10-24 10:18:02.499 13003-13003/? A/DEBUG:     #13 pc 0000000000280274  /data/data/com.google.android.gms/app_vision/ocr/libs/arm64-v8a/libocr.so
    10-24 10:18:02.499 13003-13003/? A/DEBUG:     #14 pc 0000000000067c84  /data/data/com.google.android.gms/app_vision/ocr/libs/arm64-v8a/libocr.so
    10-24 10:18:02.499 13003-13003/? A/DEBUG:     #15 pc 0000000000067f64  /data/data/com.google.android.gms/app_vision/ocr/libs/arm64-v8a/libocr.so
    10-24 10:18:02.499 13003-13003/? A/DEBUG:     #16 pc 0000000000063dec  /data/data/com.google.android.gms/app_vision/ocr/libs/arm64-v8a/libocr.so`

Android Version: **7.0**
Google Play Services Version: **11.7.43 (448-172403884)**
Project: **ocr-reader** (from Vision Samples)

**Comment by rgdixit** [comment]
+1, Consistently crashing on Galaxy S7 Edge when it tries to build `TextRecognizer`

**Comment by liuyl** [comment]
Hi, we apologize for the native crash, this is a know issue that happens on Samsung Exynos arm64 devices with Google Play services version 10.7.x. It will be fixed in the next Google Play Services release.

**Comment by LucianIacob** [comment]
@liuyl thanks for answering on this.
If you know, can you tell us the approximate date/month of the next Google Play Services release?

**Comment by Yrlec** [comment]
When will the update get released? We're currently getting thousands of crashes per day and the Google Play Android Vitals say we have have excessive crash rate, which affects our ranking. 

Any way to detect which devices have this issue so that we can disable it for the devices that are still on the buggy Google Play Services version?

**Comment by liuyl** [comment]
Here're a list of Exynos arm64 devices I generated

j7elte,hero2lte,zeroflte,herolte,j7xelte,zerolte,on7xelte,a5xelte,a3xelte,j7e3g,a5y17lte,zerofltevzw,s5neolte,on5xelte,noblelte,zenlte,j7eltetmo,zerofltespr,nobleltevzw,zeroflteatt,a3y17lte,zerofltetmo,marinelteatt,j7eltemtr,j3popeltemtr,nobleltetmo,heroltebmc,a7y17lte,zerofltebmc,a7xelte,nobleltespr,s5neoltecan,noblelteatt,zeroltetmo,zeroltevzw,zerolteatt,j5y17lte,zenltevzw,zenltetmo,hero2ltebmc,j7y17lte,j3popeltetmo,zenlteatt,zeroltespr,a5y17ltecan,zeroltebmc,j7popeltemtr,zeroflteusc,zerofltemtr,zerofltetfnvzw,j3popelteatt,j7popeltetmo,zenltespr,gtaxlwifi,j3y17lte,nobleltebmc,zeroflteaio,j7velte,a8xelte,noblelteusc,zerofltechn,j7popelteatt,j7xeltecmcc,zeroltechn,gtaxllte,xcover4lte,j3popelteaio,noblelteskt,nobleltehk,on7elte,zeroflteskt,zerolteusc,nobleltelgt,gvlteatt,nobleltektt,zeroflteacg,a7xeltextc,zerofltektt,j7xlte,herolteskt,zerofltelra,zenltebmc,j7popelteaio,hero2lteskt,zerofltelgt,zerolteskt,j5y17ltedx,a5xeltextc,heroltektt,noblelteacg,j3popeltetfntmo,zerolteacg,heroltelgt,zeroltektt,hero2ltektt,zeroltelra,zenlteusc,hero2ltelgt,nobleltelra,zeroltelgt,j3popelteue,j7xeltektt,dream2lte,a3xeltekx,a8xelteskt,a7xeltektt,dreamlte,a5y17lteskt,j7popelteue,gtanotexllte,zenlteskt,matisse10wifikx,a7xelteskt,a5y17ltektt,a7xeltelgt,gtanotexlwifikx,a5xelteskt,a5y17ltelgt,gracelte,zenltektt,xcover4ltecan,a5xeltektt,on7xelteskt,gracerlteskt,on7xeltelgt,zerofltexx,zenltelgt,a5xeltelgt,on7xeltektt,gvwifiue,dreamlteks,gracerltektt,a7xeltecmcc,a5xeltecmcc,gvltevzw,gracerltelgt,j5y17ltektt,j7popelteskt,zenltechn,nobleltechn,j5y17ltelgt,a7y17lteskt,j5y17lteskt,j7y17ltektt,gracelteskt,dream2qltesq,gtaxlltekx,gvlte,dream2qltechn,hero2ltexx,graceqltechn,gtesveltevzw,heroltexx,j3y17ltelgt,gtanotexlltekx,graceltektt,nobleltedcm,graceltelgt,j7popeltetfntmo,zeroltexx,zenltekx,dream2qltecan,zerofltectc,j5y17ltextc,nobleltejv,graceltexx,zerofltedcm,graceqltetmo,shamu,dream2lteks,zeroltesbm,J700lte,gvltexsp

**Comment by Yrlec** [comment]
Great, thanks! What's the API call to get the device name corresponding to that?

**Comment by liuyl** [comment]
Device name like Galaxy S7 edge, or like SM-G935F?

The former one won't be accurate. For example Galaxy S8+ has [multiple models](https://www.techwalls.com/samsung-galaxy-s8-sm-g950-g955-model-number-differences/), part of them uses Qualcomm chipset which shouldn't crash.

**Comment by Yrlec** [comment]
I'm just looking for a way to programmatically check the running device against the great list you assembled, so that we can disable that feature for those users.

**Comment by liuyl** [comment]
Sorry I misunderstood. That would be android.os.Build.DEVICE https://developer.android.com/reference/android/os/Build.html#DEVICE

Here're more devices that you want to disable:
F5121, gts210vewifi, a9xltechn, SO-02J, F5321, gts28vewifi, zerofltexx-user, gts210velte, ASUS_A001, CPH1611, kate, PB2PRO, F5122,

**Comment by Yrlec** [comment]
Here's the code we use to disable the feature on affected devices. 

` private static HashSet<String> deviceWithDisabledTextCheck = new HashSet<>();
    static {
        deviceWithDisabledTextCheck.add("j7elte");
        deviceWithDisabledTextCheck.add("hero2lte");
        deviceWithDisabledTextCheck.add("zeroflte");
        deviceWithDisabledTextCheck.add("herolte");
        deviceWithDisabledTextCheck.add("j7xelte");
        deviceWithDisabledTextCheck.add("zerolte");
        deviceWithDisabledTextCheck.add("on7xelte");
        deviceWithDisabledTextCheck.add("a5xelte");
        deviceWithDisabledTextCheck.add("a3xelte");
        deviceWithDisabledTextCheck.add("j7e3g");
        deviceWithDisabledTextCheck.add("a5y17lte");
        deviceWithDisabledTextCheck.add("zerofltevzw");
        deviceWithDisabledTextCheck.add("s5neolte");
        deviceWithDisabledTextCheck.add("on5xelte");
        deviceWithDisabledTextCheck.add("noblelte");
        deviceWithDisabledTextCheck.add("zenlte");
        deviceWithDisabledTextCheck.add("j7eltetmo");
        deviceWithDisabledTextCheck.add("zerofltespr");
        deviceWithDisabledTextCheck.add("nobleltevzw");
        deviceWithDisabledTextCheck.add("zeroflteatt");
        deviceWithDisabledTextCheck.add("a3y17lte");
        deviceWithDisabledTextCheck.add("zerofltetmo");
        deviceWithDisabledTextCheck.add("marinelteatt");
        deviceWithDisabledTextCheck.add("j7eltemtr");
        deviceWithDisabledTextCheck.add("j3popeltemtr");
        deviceWithDisabledTextCheck.add("nobleltetmo");
        deviceWithDisabledTextCheck.add("heroltebmc");
        deviceWithDisabledTextCheck.add("a7y17lte");
        deviceWithDisabledTextCheck.add("zerofltebmc");
        deviceWithDisabledTextCheck.add("a7xelte");
        deviceWithDisabledTextCheck.add("nobleltespr");
        deviceWithDisabledTextCheck.add("s5neoltecan");
        deviceWithDisabledTextCheck.add("noblelteatt");
        deviceWithDisabledTextCheck.add("zeroltetmo");
        deviceWithDisabledTextCheck.add("zeroltevzw");
        deviceWithDisabledTextCheck.add("zerolteatt");
        deviceWithDisabledTextCheck.add("j5y17lte");
        deviceWithDisabledTextCheck.add("zenltevzw");
        deviceWithDisabledTextCheck.add("zenltetmo");
        deviceWithDisabledTextCheck.add("hero2ltebmc");
        deviceWithDisabledTextCheck.add("j7y17lte");
        deviceWithDisabledTextCheck.add("j3popeltetmo");
        deviceWithDisabledTextCheck.add("zenlteatt");
        deviceWithDisabledTextCheck.add("zeroltespr");
        deviceWithDisabledTextCheck.add("a5y17ltecan");
        deviceWithDisabledTextCheck.add("zeroltebmc");
        deviceWithDisabledTextCheck.add("j7popeltemtr");
        deviceWithDisabledTextCheck.add("zeroflteusc");
        deviceWithDisabledTextCheck.add("zerofltemtr");
        deviceWithDisabledTextCheck.add("zerofltetfnvzw");
        deviceWithDisabledTextCheck.add("j3popelteatt");
        deviceWithDisabledTextCheck.add("j7popeltetmo");
        deviceWithDisabledTextCheck.add("zenltespr");
        deviceWithDisabledTextCheck.add("gtaxlwifi");
        deviceWithDisabledTextCheck.add("j3y17lte");
        deviceWithDisabledTextCheck.add("nobleltebmc");
        deviceWithDisabledTextCheck.add("zeroflteaio");
        deviceWithDisabledTextCheck.add("j7velte");
        deviceWithDisabledTextCheck.add("a8xelte");
        deviceWithDisabledTextCheck.add("noblelteusc");
        deviceWithDisabledTextCheck.add("zerofltechn");
        deviceWithDisabledTextCheck.add("j7popelteatt");
        deviceWithDisabledTextCheck.add("j7xeltecmcc");
        deviceWithDisabledTextCheck.add("zeroltechn");
        deviceWithDisabledTextCheck.add("gtaxllte");
        deviceWithDisabledTextCheck.add("xcover4lte");
        deviceWithDisabledTextCheck.add("j3popelteaio");
        deviceWithDisabledTextCheck.add("noblelteskt");
        deviceWithDisabledTextCheck.add("nobleltehk");
        deviceWithDisabledTextCheck.add("on7elte");
        deviceWithDisabledTextCheck.add("zeroflteskt");
        deviceWithDisabledTextCheck.add("zerolteusc");
        deviceWithDisabledTextCheck.add("nobleltelgt");
        deviceWithDisabledTextCheck.add("gvlteatt");
        deviceWithDisabledTextCheck.add("nobleltektt");
        deviceWithDisabledTextCheck.add("zeroflteacg");
        deviceWithDisabledTextCheck.add("a7xeltextc");
        deviceWithDisabledTextCheck.add("zerofltektt");
        deviceWithDisabledTextCheck.add("j7xlte");
        deviceWithDisabledTextCheck.add("herolteskt");
        deviceWithDisabledTextCheck.add("zerofltelra");
        deviceWithDisabledTextCheck.add("zenltebmc");
        deviceWithDisabledTextCheck.add("j7popelteaio");
        deviceWithDisabledTextCheck.add("hero2lteskt");
        deviceWithDisabledTextCheck.add("zerofltelgt");
        deviceWithDisabledTextCheck.add("zerolteskt");
        deviceWithDisabledTextCheck.add("j5y17ltedx");
        deviceWithDisabledTextCheck.add("a5xeltextc");
        deviceWithDisabledTextCheck.add("heroltektt");
        deviceWithDisabledTextCheck.add("noblelteacg");
        deviceWithDisabledTextCheck.add("j3popeltetfntmo");
        deviceWithDisabledTextCheck.add("zerolteacg");
        deviceWithDisabledTextCheck.add("heroltelgt");
        deviceWithDisabledTextCheck.add("zeroltektt");
        deviceWithDisabledTextCheck.add("hero2ltektt");
        deviceWithDisabledTextCheck.add("zeroltelra");
        deviceWithDisabledTextCheck.add("zenlteusc");
        deviceWithDisabledTextCheck.add("hero2ltelgt");
        deviceWithDisabledTextCheck.add("nobleltelra");
        deviceWithDisabledTextCheck.add("zeroltelgt");
        deviceWithDisabledTextCheck.add("j3popelteue");
        deviceWithDisabledTextCheck.add("j7xeltektt");
        deviceWithDisabledTextCheck.add("dream2lte");
        deviceWithDisabledTextCheck.add("a3xeltekx");
        deviceWithDisabledTextCheck.add("a8xelteskt");
        deviceWithDisabledTextCheck.add("a7xeltektt");
        deviceWithDisabledTextCheck.add("dreamlte");
        deviceWithDisabledTextCheck.add("a5y17lteskt");
        deviceWithDisabledTextCheck.add("j7popelteue");
        deviceWithDisabledTextCheck.add("gtanotexllte");
        deviceWithDisabledTextCheck.add("zenlteskt");
        deviceWithDisabledTextCheck.add("matisse10wifikx");
        deviceWithDisabledTextCheck.add("a7xelteskt");
        deviceWithDisabledTextCheck.add("a5y17ltektt");
        deviceWithDisabledTextCheck.add("a7xeltelgt");
        deviceWithDisabledTextCheck.add("gtanotexlwifikx");
        deviceWithDisabledTextCheck.add("a5xelteskt");
        deviceWithDisabledTextCheck.add("a5y17ltelgt");
        deviceWithDisabledTextCheck.add("gracelte");
        deviceWithDisabledTextCheck.add("zenltektt");
        deviceWithDisabledTextCheck.add("xcover4ltecan");
        deviceWithDisabledTextCheck.add("a5xeltektt");
        deviceWithDisabledTextCheck.add("on7xelteskt");
        deviceWithDisabledTextCheck.add("gracerlteskt");
        deviceWithDisabledTextCheck.add("on7xeltelgt");
        deviceWithDisabledTextCheck.add("zerofltexx");
        deviceWithDisabledTextCheck.add("zenltelgt");
        deviceWithDisabledTextCheck.add("a5xeltelgt");
        deviceWithDisabledTextCheck.add("on7xeltektt");
        deviceWithDisabledTextCheck.add("gvwifiue");
        deviceWithDisabledTextCheck.add("dreamlteks");
        deviceWithDisabledTextCheck.add("gracerltektt");
        deviceWithDisabledTextCheck.add("a7xeltecmcc");
        deviceWithDisabledTextCheck.add("a5xeltecmcc");
        deviceWithDisabledTextCheck.add("gvltevzw");
        deviceWithDisabledTextCheck.add("gracerltelgt");
        deviceWithDisabledTextCheck.add("j5y17ltektt");
        deviceWithDisabledTextCheck.add("j7popelteskt");
        deviceWithDisabledTextCheck.add("zenltechn");
        deviceWithDisabledTextCheck.add("nobleltechn");
        deviceWithDisabledTextCheck.add("j5y17ltelgt");
        deviceWithDisabledTextCheck.add("a7y17lteskt");
        deviceWithDisabledTextCheck.add("j5y17lteskt");
        deviceWithDisabledTextCheck.add("j7y17ltektt");
        deviceWithDisabledTextCheck.add("gracelteskt");
        deviceWithDisabledTextCheck.add("dream2qltesq");
        deviceWithDisabledTextCheck.add("gtaxlltekx");
        deviceWithDisabledTextCheck.add("gvlte");
        deviceWithDisabledTextCheck.add("dream2qltechn");
        deviceWithDisabledTextCheck.add("hero2ltexx");
        deviceWithDisabledTextCheck.add("graceqltechn");
        deviceWithDisabledTextCheck.add("gtesveltevzw");
        deviceWithDisabledTextCheck.add("heroltexx");
        deviceWithDisabledTextCheck.add("j3y17ltelgt");
        deviceWithDisabledTextCheck.add("gtanotexlltekx");
        deviceWithDisabledTextCheck.add("graceltektt");
        deviceWithDisabledTextCheck.add("nobleltedcm");
        deviceWithDisabledTextCheck.add("graceltelgt");
        deviceWithDisabledTextCheck.add("j7popeltetfntmo");
        deviceWithDisabledTextCheck.add("zeroltexx");
        deviceWithDisabledTextCheck.add("zenltekx");
        deviceWithDisabledTextCheck.add("dream2qltecan");
        deviceWithDisabledTextCheck.add("zerofltectc");
        deviceWithDisabledTextCheck.add("j5y17ltextc");
        deviceWithDisabledTextCheck.add("nobleltejv");
        deviceWithDisabledTextCheck.add("graceltexx");
        deviceWithDisabledTextCheck.add("zerofltedcm");
        deviceWithDisabledTextCheck.add("graceqltetmo");
        deviceWithDisabledTextCheck.add("shamu");
        deviceWithDisabledTextCheck.add("dream2lteks");
        deviceWithDisabledTextCheck.add("zeroltesbm");
        deviceWithDisabledTextCheck.add("j700lte");
        deviceWithDisabledTextCheck.add("gvltexsp");
        deviceWithDisabledTextCheck.add("f5121");
        deviceWithDisabledTextCheck.add("gts210vewifi");
        deviceWithDisabledTextCheck.add("a9xltechn");
        deviceWithDisabledTextCheck.add("so-02j");
        deviceWithDisabledTextCheck.add("f5321");
        deviceWithDisabledTextCheck.add("gts28vewifi");
        deviceWithDisabledTextCheck.add("zerofltexx-user");
        deviceWithDisabledTextCheck.add("gts210velte");
        deviceWithDisabledTextCheck.add("asus_a001");
        deviceWithDisabledTextCheck.add("cph1611");
        deviceWithDisabledTextCheck.add("kate");
        deviceWithDisabledTextCheck.add("pb2pro");
        deviceWithDisabledTextCheck.add("f5122");
    }

    private static boolean shouldDisableTextCheck() {
        String deviceName = android.os.Build.DEVICE;
        if(Util.isNullOrBlank(deviceName)) {
            return false;
        }
        deviceName = Util.toLowerCaseInvariant(deviceName);
        // Some devices crash on a certain version of Google Play Services.
        // https://github.com/googlesamples/android-vision/issues/269#issuecomment-339464902
        // We therefore check for it here.
        return deviceWithDisabledTextCheck.contains(deviceName);
    }`

**Comment by ankitbatra11** [comment]
I am seeing it on Asus ZenFone 5 (ASUS_T00J) as well. Here is the backtrace:
```
 native: pc 00000000001b66f6  /data/data/com.google.android.gms/app_vision/ocr/libs/x86/libocr.so
  native: pc 00000000002632cd  /data/data/com.google.android.gms/app_vision/ocr/libs/x86/libocr.so
  native: pc 000000000026390e  /data/data/com.google.android.gms/app_vision/ocr/libs/x86/libocr.so
  native: pc 00000000001bc573  /data/data/com.google.android.gms/app_vision/ocr/libs/x86/libocr.so
  native: pc 00000000001bda57  /data/data/com.google.android.gms/app_vision/ocr/libs/x86/libocr.so
  native: pc 000000000002f1af  /data/data/com.google.android.gms/app_vision/ocr/libs/x86/libocr.so
  native: pc 000000000002ed01  /data/data/com.google.android.gms/app_vision/ocr/libs/x86/libocr.so
  native: pc 00000000001c21c5  /data/data/com.google.android.gms/app_vision/ocr/libs/x86/libocr.so
  native: pc 00000000001c45a2  /data/data/com.google.android.gms/app_vision/ocr/libs/x86/libocr.so
  native: pc 00000000000317b2  /data/data/com.google.android.gms/app_vision/ocr/libs/x86/libocr.so
  native: pc 0000000000031a62  /data/data/com.google.android.gms/app_vision/ocr/libs/x86/libocr.so
  native: pc 0000000000031b25  /data/data/com.google.android.gms/app_vision/ocr/libs/x86/libocr.so
  native: pc 000000000002d851  /data/data/com.google.android.gms/app_vision/ocr/libs/x86/libocr.so
  native: pc 0000000000083585  /system/lib/libdvm.so
  native: pc 0000000000002e5f  [anon:libc_malloc]
  native: pc 00000000000e094c  /system/lib/libdvm.so
  native: pc 0000000000237db9  /system/lib/libdvm.so
  native: pc 00000000000057a3  <unknown>
  native: pc 0000000000098ec0  /system/lib/libdvm.so
  native: pc 0000000000094b01  /system/lib/libdvm.so
  native: pc 00000000001139b8  /system/lib/libdvm.so
  native: pc 0000000000126e89  /system/lib/libdvm.so
  native: pc 0000000000237db9  /system/lib/libdvm.so
  native: pc 0000000000005e9f  <unknown>
  native: pc 0000000000098ec0  /system/lib/libdvm.so
  native: pc 0000000000094b01  /system/lib/libdvm.so
  native: pc 00000000001146ce  /system/lib/libdvm.so
  native: pc 00000000000d4a1d  /system/lib/libdvm.so
  native: pc 000000000005f93a  /system/lib/libandroid_runtime.so
  native: pc 000000000006121c  /system/lib/libandroid_runtime.so
  native: pc 0000000000000f1c  /system/bin/app_process
  native: pc 000000000001dd7a  /system/lib/libc.so
```

**Comment by GEllickson** [comment]
@liuyl Thanks for getting back to us. Can you confirm if this affects just text recognition or other detectors as well? We use both text and barcode detection, so we paused both off after the bug, but we'd like to still scan for barcodes if that won't crash anything.

**Comment by liuyl** [comment]
This only affects text recognition API.

**Comment by radheshyamsingh24** [comment]
I am using sdk for text recognition in my app and getting same issue. I got lots of negative review due to this crash. Please let us know whenever this issue will be resolved.

**Comment by minyoungjun** [comment]
I have same problem in Galaxy S7 edge. Please fix it as soon as possible. It kills me.

**Comment by nhattv18** [comment]
I have same problem. Many users have uninstalled and 1 star rating :( Please fix it soon. Thanks!

**Comment by balrampandey19** [comment]
I Have same issue on Honor phone, any alternative solution for this problem?.


11-06 15:15:04.859 30810-30810/? A/DEBUG: signal 6 (SIGABRT), code -6 (SI_TKILL), fault addr --------
11-06 15:15:04.861 30810-30810/? A/DEBUG: Abort message: 'timer.cc:38 Check failed: x >= 1 (0 vs. 1) 
                                          '
11-06 15:15:04.862 30810-30810/? A/DEBUG:     x0   0000000000000000  x1   00000000000077a9  x2   0000000000000006  x3   0000000000000008
11-06 15:15:04.862 30810-30810/? A/DEBUG:     x4   0000000000000073  x5   0000008000000000  x6   0000007e4a513000  x7   0000000000000000
11-06 15:15:04.862 30810-30810/? A/DEBUG:     x8   0000000000000083  x9   ffffffffffffffdf  x10  0000000000000000  x11  0000000000000001
11-06 15:15:04.862 30810-30810/? A/DEBUG:     x12  ffffffffffffffff  x13  0000000000000000  x14  0000000000000000  x15  000b942cb9043e02
11-06 15:15:04.862 30810-30810/? A/DEBUG:     x16  0000007e47a22ec8  x17  0000007e479cbc28  x18  0000007e3b3e09ad  x19  0000007e4a5c6b40
11-06 15:15:04.862 30810-30810/? A/DEBUG:     x20  0000000000000006  x21  0000007e4a5c6a98  x22  0000000000000002  x23  0000007e36546f40
11-06 15:15:04.862 30810-30810/? A/DEBUG:     x24  0000007e11cf2780  x25  0000007e21dd22c0  x26  0000007e11617478  x27  0000007fc1e52928
11-06 15:15:04.862 30810-30810/? A/DEBUG:     x28  0000000000000000  x29  0000007fc1e51ab0  x30  0000007e479c90d0
11-06 15:15:04.862 30810-30810/? A/DEBUG:     sp   0000007fc1e51a90  pc   0000007e479cbc30  pstate 0000000060000000
11-06 15:15:04.867 30810-30810/? A/DEBUG: backtrace:
11-06 15:15:04.867 30810-30810/? A/DEBUG:     #00 pc 000000000006bc30  /system/lib64/libc.so (tgkill+8)
11-06 15:15:04.867 30810-30810/? A/DEBUG:     #01 pc 00000000000690cc  /system/lib64/libc.so (pthread_kill+64)
11-06 15:15:04.867 30810-30810/? A/DEBUG:     #02 pc 0000000000023e68  /system/lib64/libc.so (raise+24)
11-06 15:15:04.867 30810-30810/? A/DEBUG:     #03 pc 000000000001c8ec  /system/lib64/libc.so (abort+52)
11-06 15:15:04.867 30810-30810/? A/DEBUG:     #04 pc 00000000004750d0  /data/data/com.google.android.gms/app_vision/ocr/libs/arm64-v8a/libocr.so
11-06 15:15:04.867 30810-30810/? A/DEBUG:     #05 pc 0000000000475770  /data/data/com.google.android.gms/app_vision/ocr/libs/arm64-v8a/libocr.so
11-06 15:15:04.867 30810-30810/? A/DEBUG:     #06 pc 000000000047583c  /data/data/com.google.android.gms/app_vision/ocr/libs/arm64-v8a/libocr.so
11-06 15:15:04.867 30810-30810/? A/DEBUG:     #07 pc 0000000000474830  /data/data/com.google.android.gms/app_vision/ocr/libs/arm64-v8a/libocr.so
11-06 15:15:04.867 30810-30810/? A/DEBUG:     #08 pc 000000000026cee4  /data/data/com.google.android.gms/app_vision/ocr/libs/arm64-v8a/libocr.so
11-06 15:15:04.867 30810-30810/? A/DEBUG:     #09 pc 0000000000271190  /data/data/com.google.android.gms/app_vision/ocr/libs/arm64-v8a/libocr.so
11-06 15:15:04.867 30810-30810/? A/DEBUG:     #10 pc 0000000000065794  /data/data/com.google.android.gms/app_vision/ocr/libs/arm64-v8a/libocr.so
11-06 15:15:04.867 30810-30810/? A/DEBUG:     #11 pc 0000000000065488  /data/data/com.google.android.gms/app_vision/ocr/libs/arm64-v8a/libocr.so
11-06 15:15:04.867 30810-30810/? A/DEBUG:     #12 pc 000000000027caf4  /data/data/com.google.android.gms/app_vision/ocr/libs/arm64-v8a/libocr.so
11-06 15:15:04.867 30810-30810/? A/DEBUG:     #13 pc 0000000000280274  /data/data/com.google.android.gms/app_vision/ocr/libs/arm64-v8a/libocr.so
11-06 15:15:04.867 30810-30810/? A/DEBUG:     #14 pc 0000000000067c84  /data/data/com.google.android.gms/app_vision/ocr/libs/arm64-v8a/libocr.so
11-06 15:15:04.867 30810-30810/? A/DEBUG:     #15 pc 0000000000067f64  /data/data/com.google.android.gms/app_vision/ocr/libs/arm64-v8a/libocr.so
11-06 15:15:04.867 30810-30810/? A/DEBUG:     #16 pc 0000000000063dec  /data/data/com.google.android.gms/app_vision/ocr/libs/arm64-v8a/libocr.so (Java_com_google_android_libraries_vision_ocr_NativeOcr_nativeInitializeWithTypeAndResult+96)
11-06 15:15:04.867 30810-30810/? A/DEBUG:     #17 pc 0000000000871d10  /data/user_de/0/com.google.android.gms/app_chimera/m/0000000b/oat/arm64/DynamiteModulesA_GmsCore_prodmnc_xxhdpi_release.odex (offset 0x4e8000)
11-06 15:15:04.902 1715-12264/? I/DownloadState: @@@@@@@@@@@@ uid :10156 download :true

**Comment by dvelasquezBelatrix** [comment]
I have the same issue on galaxy S6 running Android 6.1 and one plus 5 running 7.1.
Those crashes started today!

**Comment by leandroflores2000** [comment]
I started having the same issue today with Play Services version 11.7.45 (448-174122868) on a Galaxy S7 (SM-G930F) Android Version 7.0

**Comment by LukeEdghar** [comment]
I am facing the same problem while instantiating TextRecognizer object. Is there any way to prevent the app crash if the device doesn't support the TextRecognizer? I was expecting something like a try-catch to handle the problem.

Thanks in advance

**Comment by geoffduong** [comment]
Getting this error as well on Samsung Galaxy S6

```W/DynamiteModule: Local module descriptor class for com.google.android.gms.vision.dynamite not found.
I/DynamiteModule: Considering local module com.google.android.gms.vision.dynamite:0 and remote module com.google.android.gms.vision.dynamite:1402
I/DynamiteModule: Selected remote version of com.google.android.gms.vision.dynamite, version >= 1402
W/System: ClassLoader referenced unknown path: 
W/System: ClassLoader referenced unknown path: /data/user_de/0/com.google.android.gms/app_chimera/m/00000037/n/arm64-v8a
I/Vision: Loading library libocr.so
I/Vision: libocr.so library load status: true
I/Vision: Reading ocr models from /data/user/0/com.google.android.gms/app_vision/ocr/data/models
I/native: jni_helper.cc:166 Loading model config /data/user/0/com.google.android.gms/app_vision/ocr/data/models/semanticlift_engine_0.2.bincfg
E/native: jni_helper.cc:170 GetContents failed: /data/user/0/com.google.android.gms/app_vision/ocr/data/models/semanticlift_engine_0.2.bincfg
I/native: jni_helper.cc:166 Loading model config /data/user/0/com.google.android.gms/app_vision/ocr/data/models/tfmini_lstm_engine_latin_0.2.bincfg
A/native: timer.cc:38 Check failed: x >= 1 (0 vs. 1) 
A/native: terminating.

**Comment by liuyl** [comment]
I am really sorry for the crash, please refer to Yrlec's comment about how to disable textRecognizer on devices that may crash.

**Comment by LukeEdghar** [comment]
Is it possible to use an older version of the google play service library? I have tried to use an older version  declaring in my gradle module an old dependency, but I believe that the app is still using the newest one, delivered by the Google Play Service application

**Comment by barmic12** [comment]
The same problem on Huawei P9. Is there any solution to resolve this problem (not disable text recognition)?

**Comment by evis94** [comment]
@barmic12 Also on Huawei P9 plus with these details:

System on Chip
HiSilicon KIRIN955 (Kirin 955)

CPU
4x ARM Cortex-A72 (2500 Mhz)
4x ARM Cortex-A53 (1800 Mhz)

GPU
ARM Mali T880 (900 Mhz)

@liuyl We have done the suggested excluding of devices but there seem to be more, all of which are demoting the rating of the application to 1 star. Any chance you have a list with more affected devices?

**Comment by evis94** [comment]
If it helps I've decided to map P9 devices according to the Google play Supported devices and exclude them as well:
P9: HWEVA 
P9 Lite premium: HWVNS-Q
P9 plus: HWVIE
P9 Lite: HWVNS-H
P9 lite smart: HWDIG-L8940

Hope that Google will fix before we will have more surprises.

**Comment by lodenrogue** [comment]
This stopped working for us all of a sudden. It was working a few weeks ago. We didn't change anything. Why would that be?

**Comment by nhattv18** [comment]
@lodenrogue Me too, my samsung s8 (dreamlte) was working a month ago, but today does not work :(

**Comment by evis94** [comment]
More devices are crashing on this with latest Google play console crash reports: 
HUAWEI Mate 10 lite: HWRNE
Huawei Mate 9 (HWMHA),
Huawei P10 lite (HWWAS-H)
Honor 8 (HWFRD)
P10 (HWVTR)
HUAWEI nova 2 Plus (HWBAC)
HUAWEI GR5 2017 (HWBLN-H)
M3 (hwbeethoven)
Xperia X (F5121)

**Comment by skeeping** [comment]
Do we have an ETA on the next Google Play Services release this is affecting an upcoming release for my team.

**Comment by raulsb** [comment]
Day by day its bigger the number of crashing devices, is there a way for a downgrade? please let us know about an ETA on the next Google Play Services release

**Comment by wanwjing** [comment]
Hi all, 

I also meet up on this error in this a few days. 

@Yrlec thx for your suggestion.

crash device:- 
Samsung S7, Android 7.0
Samsung 8 plus Android 7.0 
Huawei P9 lite Android 6.0 

Error log :-
11-08 17:19:47.985 19278-19278/? E/System: java.lang.Runtime#loadLibrary(String, ClassLoader) is private and will be removed in a future Android release
11-08 17:19:47.990 19278-19278/? E/System: java.lang.Runtime#loadLibrary(String, ClassLoader) is private and will be removed in a future Android release
11-08 17:19:47.999 19278-19278/? E/ActivityThread: Failed to find provider info for com.tencent.mm.coolassist.debugprovider
11-08 17:19:48.033 17423-17423/com.silverlake.simulator E/native: jni_helper.cc:170 GetContents failed: /data/user/0/com.google.android.gms/app_vision/ocr/data/models/semanticlift_engine_0.2.bincfg
11-08 17:19:48.034 17423-17423/com.silverlake.simulator A/native: timer.cc:38 Check failed: x >= 1 (0 vs. 1) 
11-08 17:19:48.034 17423-17423/com.silverlake.simulator A/native: terminating.
11-08 17:19:48.035 17423-17423/com.silverlake.simulator A/libc: Fatal signal 6 (SIGABRT), code -6 in tid 17423 (rlake.simulator)
                                                                
                                                                [ 11-08 17:19:48.038  3135: 3135 W/         ]
                                                                debuggerd: handling request: pid=17423 uid=10316 gid=10316 tid=17423
11-08 17:19:48.062 3335-3335/? E/audit: type=1400 audit(1510132788.054:121674): avc:  denied  { search } for  pid=19316 comm="debuggerd64" name="com.google.android.gms" dev="dm-1" ino=1114223 scontext=u:r:debuggerd:s0 tcontext=u:object_r:app_data_file:s0:c512,c768 tclass=dir permissive=0 SEPF_SECMOBILE_7.0_0009 audit_filtered
11-08 17:19:48.062 3335-3335/? E/audit: type=1300 audit(1510132788.054:121674): arch=c00000b7 syscall=56 success=no exit=-13 a0=ffffff9c a1=7d3108b480 a2=0 a3=0 items=0 ppid=3135 pid=19316 auid=4294967295 uid=0 gid=0 euid=0 suid=0 fsuid=0 egid=0 sgid=0 fsgid=0 tty=(none) ses=4294967295 comm="debuggerd64" exe="/system/bin/debuggerd64" subj=u:r:debuggerd:s0 key=(null)
11-08 17:19:48.062 3335-3335/? E/audit: type=1327 audit(1510132788.054:121674): proctitle="/system/bin/debuggerd64"
11-08 17:19:48.062 3335-3335/? E/audit: type=1400 audit(1510132788.054:121675): avc:  denied  { search } for  pid=19316 comm="debuggerd64" name="com.google.android.gms" dev="dm-1" ino=1376452 scontext=u:r:debuggerd:s0 tcontext=u:object_r:app_data_file:s0:c512,c768 tclass=dir permissive=0 SEPF_SECMOBILE_7.0_0009 audit_filtered
11-08 17:19:48.062 3335-3335/? E/audit: type=1300 audit(1510132788.054:121675): arch=c00000b7 syscall=56 success=no exit=-13 a0=ffffff9c a1=7d31026380 a2=0 a3=0 items=0 ppid=3135 pid=19316 auid=4294967295 uid=0 gid=0 euid=0 suid=0 fsuid=0 egid=0 sgid=0 fsgid=0 tty=(none) ses=4294967295 comm="debuggerd64" exe="/system/bin/debuggerd64" subj=u:r:debuggerd:s0 key=(null)
11-08 17:19:48.062 3335-3335/? E/audit: type=1327 audit(1510132788.054:121675): proctitle="/system/bin/debuggerd64"
11-08 17:19:48.147 19278-19278/? E/System: java.lang.Runtime#loadLibrary(String, ClassLoader) is private and will be removed in a future Android release
11-08 17:19:48.199 19316-19316/? A/DEBUG: *** *** *** *** *** *** *** *** *** *** *** *** *** *** *** ***
11-08 17:19:48.200 19316-19316/? A/DEBUG: Build fingerprint: 'samsung/dream2ltexx/dream2lte:7.0/NRD90M/G955FXXU1AQH3:user/release-keys'
11-08 17:19:48.200 19316-19316/? A/DEBUG: Revision: '10'
11-08 17:19:48.200 19316-19316/? A/DEBUG: ABI: 'arm64'
11-08 17:19:48.200 19316-19316/? A/DEBUG: pid: 17423, tid: 17423, name: rlake.simulator  >>> com.silverlake.simulator <<<
11-08 17:19:48.200 19316-19316/? A/DEBUG: signal 6 (SIGABRT), code -6 (SI_TKILL), fault addr --------
11-08 17:19:48.206 19316-19316/? A/DEBUG: Abort message: 'timer.cc:38 Check failed: x >= 1 (0 vs. 1) 
                                          '
11-08 17:19:48.206 19316-19316/? A/DEBUG:     x0   0000000000000000  x1   000000000000440f  x2   0000000000000006  x3   0000000000000008
11-08 17:19:48.206 19316-19316/? A/DEBUG:     x4   00000000000000f7  x5   0000008000000000  x6   000000730eab7000  x7   0000000000000000
11-08 17:19:48.206 19316-19316/? A/DEBUG:     x8   0000000000000083  x9   ffffffffffffffdf  x10  0000000000000000  x11  ffffffffffffffff
11-08 17:19:48.206 19316-19316/? A/DEBUG:     x12  0000000000000018  x13  0000000000000000  x14  0000000000000000  x15  000213a984fb1e14
11-08 17:19:48.206 19316-19316/? A/DEBUG:     x16  000000730c293ed0  x17  000000730c23b9f4  x18  000000730a4ee8ed  x19  000000730eb70b40
11-08 17:19:48.206 19316-19316/? A/DEBUG:     x20  0000000000000006  x21  000000730eb70a98  x22  0000000000000002  x23  00000072dd0ae520
11-08 17:19:48.206 19316-19316/? A/DEBUG:     x24  00000072d79f3780  x25  00000072cd959d40  x26  00000072d7318478  x27  0000007fe76fe4c8
11-08 17:19:48.206 19316-19316/? A/DEBUG:     x28  0000000000000000  x29  0000007fe76fd650  x30  000000730c238d14
11-08 17:19:48.206 19316-19316/? A/DEBUG:     sp   0000007fe76fd630  pc   000000730c23b9fc  pstate 0000000060000000
11-08 17:19:48.209 3335-3335/? E/audit: type=1400 audit(1510132788.202:121676): avc:  denied  { search } for  pid=19316 comm="debuggerd64" name="com.google.android.gms" dev="dm-1" ino=1114223 scontext=u:r:debuggerd:s0 tcontext=u:object_r:app_data_file:s0:c512,c768 tclass=dir permissive=0 SEPF_SECMOBILE_7.0_0009 audit_filtered
11-08 17:19:48.209 3335-3335/? E/audit: type=1300 audit(1510132788.202:121676): arch=c00000b7 syscall=56 success=no exit=-13 a0=ffffff9c a1=7d3108b480 a2=0 a3=0 items=0 ppid=3135 pid=19316 auid=4294967295 uid=1045 gid=1045 euid=1045 suid=1045 fsuid=1045 egid=1045 sgid=1045 fsgid=1045 tty=(none) ses=4294967295 comm="debuggerd64" exe="/system/bin/debuggerd64" subj=u:r:debuggerd:s0 key=(null)
11-08 17:19:48.209 3335-3335/? E/audit: type=1327 audit(1510132788.202:121676): proctitle="/system/bin/debuggerd64"
11-08 17:19:48.217 3335-3335/? E/audit: type=1400 audit(1510132788.210:121677): avc:  denied  { search } for  pid=19316 comm="debuggerd64" name="com.google.android.gms" dev="dm-1" ino=1376452 scontext=u:r:debuggerd:s0 tcontext=u:object_r:app_data_file:s0:c512,c768 tclass=dir permissive=0 SEPF_SECMOBILE_7.0_0009 audit_filtered
11-08 17:19:48.217 3335-3335/? E/audit: type=1300 audit(1510132788.210:121677): arch=c00000b7 syscall=56 success=no exit=-13 a0=ffffff9c a1=7d31026380 a2=0 a3=0 items=0 ppid=3135 pid=19316 auid=4294967295 uid=1045 gid=1045 euid=1045 suid=1045 fsuid=1045 egid=1045 sgid=1045 fsgid=1045 tty=(none) ses=4294967295 comm="debuggerd64" exe="/system/bin/debuggerd64" subj=u:r:debuggerd:s0 key=(null)
11-08 17:19:48.217 3335-3335/? E/audit: type=1327 audit(1510132788.210:121677): proctitle="/system/bin/debuggerd64"
11-08 17:19:48.217 19316-19316/? A/DEBUG: backtrace:
11-08 17:19:48.218 19316-19316/? A/DEBUG:     #00 pc 000000000006f9fc  /system/lib64/libc.so (tgkill+8)
11-08 17:19:48.218 19316-19316/? A/DEBUG:     #01 pc 000000000006cd10  /system/lib64/libc.so (pthread_kill+64)
11-08 17:19:48.218 19316-19316/? A/DEBUG:     #02 pc 0000000000025078  /system/lib64/libc.so (raise+24)
11-08 17:19:48.218 19316-19316/? A/DEBUG:     #03 pc 000000000001cc04  /system/lib64/libc.so (abort+52)
11-08 17:19:48.218 19316-19316/? A/DEBUG:     #04 pc 00000000004750d0  /data/data/com.google.android.gms/app_vision/ocr/libs/arm64-v8a/libocr.so
11-08 17:19:48.218 19316-19316/? A/DEBUG:     #05 pc 0000000000475770  /data/data/com.google.android.gms/app_vision/ocr/libs/arm64-v8a/libocr.so
11-08 17:19:48.218 19316-19316/? A/DEBUG:     #06 pc 0000000000475814  /data/data/com.google.android.gms/app_vision/ocr/libs/arm64-v8a/libocr.so
11-08 17:19:48.218 19316-19316/? A/DEBUG:     #07 pc 0000000000474830  /data/data/com.google.android.gms/app_vision/ocr/libs/arm64-v8a/libocr.so
11-08 17:19:48.218 19316-19316/? A/DEBUG:     #08 pc 000000000026cee4  /data/data/com.google.android.gms/app_vision/ocr/libs/arm64-v8a/libocr.so
11-08 17:19:48.218 19316-19316/? A/DEBUG:     #09 pc 0000000000271190  /data/data/com.google.android.gms/app_vision/ocr/libs/arm64-v8a/libocr.so
11-08 17:19:48.218 19316-19316/? A/DEBUG:     #10 pc 0000000000065794  /data/data/com.google.android.gms/app_vision/ocr/libs/arm64-v8a/libocr.so
11-08 17:19:48.218 19316-19316/? A/DEBUG:     #11 pc 0000000000065488  /data/data/com.google.android.gms/app_vision/ocr/libs/arm64-v8a/libocr.so
11-08 17:19:48.218 19316-19316/? A/DEBUG:     #12 pc 000000000027caf4  /data/data/com.google.android.gms/app_vision/ocr/libs/arm64-v8a/libocr.so
11-08 17:19:48.218 19316-19316/? A/DEBUG:     #13 pc 0000000000280274  /data/data/com.google.android.gms/app_vision/ocr/libs/arm64-v8a/libocr.so
11-08 17:19:48.218 19316-19316/? A/DEBUG:     #14 pc 0000000000067c84  /data/data/com.google.android.gms/app_vision/ocr/libs/arm64-v8a/libocr.so
11-08 17:19:48.218 19316-19316/? A/DEBUG:     #15 pc 0000000000067f64  /data/data/com.google.android.gms/app_vision/ocr/libs/arm64-v8a/libocr.so
11-08 17:19:48.218 19316-19316/? A/DEBUG:     #16 pc 0000000000063dec  /data/data/com.google.android.gms/app_vision/ocr/libs/arm64-v8a/libocr.so
11-08 17:19:48.218 19316-19316/? A/DEBUG:     #17 pc 0000000000387e60  /data/user_de/0/com.google.android.gms/app_chimera/m/00000023/oat/arm64/DynamiteModulesA_GmsCore_prodmnc_alldpi_release.odex (offset 0x4e7000)
11-08 17:19:48.755 3797-5289/? E/SensorService: getSensorList a sensor (Rear Ambient Light) without holding its required permission: com.samsung.permission.HRM_EXT
11-08 17:19:48.756 3797-5289/? E/SensorService: getSensorList a sensor (Rear Proximity detecting) without holding its required permission: com.samsung.permission.HRM_EXT
11-08 17:19:48.756 3797-5289/? E/SensorService: getSensorList a sensor (HRMLED IR) without holding its required permission: android.permission.BODY_SENSORS
11-08 17:19:48.756 3797-5289/? E/SensorService: getSensorList a sensor (HRMLED RED) without holding its required permission: android.permission.BODY_SENSORS
11-08 17:19:48.756 3797-5289/? E/SensorService: getSensorList a sensor (ADPD143RI) without holding its required permission: com.samsung.permission.SSENSOR
11-08 17:19:48.756 3797-5289/? E/SensorService: getSensorList a sensor (Proximity Alert Sensor) without holding its required permission: com.samsung.permission.SSENSOR
11-08 17:19:48.756 3797-5289/? E/SensorService: getSensorList a sensor (HRM Sensor) without holding its required permission: com.samsung.permission.SSENSOR
11-08 17:19:48.757 3797-5289/? E/SensorService: getSensorList a sensor (HeartRate Sensor) without holding its required permission: android.permission.BODY_SENSORS
11-08 17:19:48.757 3797-5289/? E/SensorService: getSensorList a sensor (Sensor Diagnostic Monitor) without holding its required permission: com.samsung.permission.SSENSOR

**Comment by PlayableApps** [comment]
getting the same error please help us asap.

**Comment by PlayableApps** [comment]
Guys there's a new release out there now 11.6.0 but i'm not sure if they fixed this issue or not yet.

**Comment by Yrlec** [comment]
After this we also need a post mortem. How can a trivially reproducible bug get through QA? If Google just ran a very basic unit test on all CPU architectures they would have caught this. Given the number of apps and devices using this library, how come that has not been done? How come the faulty version gets rolled out to more devices after the bug has been identified?

**Comment by Cliffordwh** [comment]
did that latest push of there fix this issue? Man this is effecting us badly

**Comment by radunicolescu** [comment]
There are 2 different play services and this might increase the confusion:
1. Android Play Services library: latest version **11.6.0** released on November 6, 2017
2. Android Play Services app installable from play store: latest version **11.7.45** released on November 2 2017
On mobile vision api documentation (https://developers.google.com/vision/text-overview) google added the warning

`On Google Play Services version 10.7.x you may see crashes when using Text API on Samsung Exynos arm64 devices. This is a known issue and will be fixed in the next release.`

Since both the play services application and the play services library are well above the 10.7.x version and this issue bug is still there I think what we actually wait for is a new release of the google vision library that fixes this bug.

Google Vision library releases can be followed here
https://developers.google.com/vision/android/release-notes

I am also waiting for a fix on this issue. Maybe there is someone that can give us a rough estimation about when this is going to be fixed. If the fix will take a long time maybe some developers will choose to switch to alternative solutions.

**Comment by Cliffordwh** [comment]
So from a App development side, there is nothing we can really do other than wait for users to update there devices?

**Comment by radheshyamsingh24** [comment]
From App side we can do one thing which will minimize crashes. We can write code to track if app is crashing on device during instantiation of TextRecognizer object and then from next time we can bypass/disable OCR feature. But on other devices(on which there is no crash) it will work perfectly.

This will lead to crash once on impacted devices and from next time no crash(Yes, it will impact feature). 

May be this will help little bit. Thanks

**Comment by radunicolescu** [comment]
@radheshyamsingh24 there is no possibility to detect a native crash. If you try to put it in a `try catch` block in java the exception will not be caught.
So the only real option is to use a fixed list of devices that will be excluded as describer by @Yrlec but this is far from perfect since you can have the exact same device type with different type of CPUs so you will end up blocking devices that would not crash. Also nobody can make a 100% complete list of devices that might have Exynos CPU so there is always the chance of new crashes on devices that are not in the list.

**Comment by PlayableApps** [comment]
so this release 11.6.0 did not fix the issue, i guess we have to wait at least a month till the next release comes .

**Comment by radheshyamsingh24** [comment]
@radunicolescu : Yes we can't detect native crash in try catch block and device list was also not much helpful in my case. Thats why  i have used Preference concept here. 

boolean isFirstTimeUser = getBooleanFromPreference (default value TRUE) 
boolean isGVSupportThisDevice = getBooleanFromPreference (default value False)

 if (isFirstTimeUser || isGVSupportThisDevice) {
     Update Preference to make isFirstTimeUser to false

     textRecognizer = new TextRecognizer.Builder(context).build();

      Update Preference to make isGVSupportThisDevice to false    
}

Hope this code will help you.

**Comment by badoualy** [comment]
Joining the queue to wait a fix for this :)
For ref:
```
荣耀8青春版 (HWPRA-H) | 10 | 32.3%
P10 lite (HWWAS-H) | 8 | 25.8%
Honor 9 (HWSTF) | 7 | 22.6%
Galaxy A5(2017) (a5y17lte) | 6 | 19.4%
```

**Comment by PlayableApps** [comment]
i don't know why carshes numbers is getting bigger and bigger! because i was only seeing few crashes on few devices !

**Comment by bibek011** [comment]
I also had this exact same problem and fixed it by using lower version of vision 
`compile 'com.google.android.gms:play-services-vision:10.0.0'`

but since today started getting the error again.
has anyone found any fix for this yet?

**Comment by nhattv18** [comment]
@bibek011 I have tried all versions and are currently using version 10.0.0 but still not working

**Comment by yonifine** [comment]
Facing the same issue here as well. I was working a couple of days back and all of a sudden my app starts to crash

**Comment by bibek011** [comment]
looks like Clearing All Data of Google Play service worked for me at least for now.

**Comment by yonifine** [comment]
@bibek011 thanks, looks it worked for me as well after clearing Google Play services.

**Comment by barmic12** [comment]
@bibek011, works for me too, thanks man!

**Comment by zebpaypeswani** [comment]
We are adding the device list in firebase. if more devices are crashing then we will manually add them in firebase.

**Comment by lodenrogue** [comment]
I feel like Google needs to put out an emergency fix for this. Please do something!

**Comment by barmic12** [comment]
Clearing All Data from Google Play services helps only for a moment. During on first run, app starts, if I run again, unfortunatelly it crash...

**Comment by bibek011** [comment]
@barmic12 for me it's still working. which google play service version are you using. i am on 11.9.45 2hich release d yesterday.

**Comment by PlayableApps** [comment]
@bibek011 latest version is 11.6.0 i believe!

**Comment by bibek011** [comment]
@PlayableApps i have joined the beta tester for the Google Play Service maybe that's why i have 11.9.45

**Comment by evis94** [comment]
Any progress? the list of devices crashing is getting bigger and bigger each day. A lot of the users are rating our app with a 1 star vain.

**Comment by nhattv18** [comment]
Too long time to fix the problem :( We are still looking forward to the new version.

**Comment by amiraelsayed** [comment]
hello I am facing the same problem, the ocr library was working with me on android 7.0 couple of days ago and suddenly I try to open it today to find this native error, please solve it as soon as you can

**Comment by liuyl** [comment]
Ask bibek011@ mentioned, the new Google Play Service version with OCR fix is currently in Beta channel.

**Comment by raulsb** [comment]
Only in beta Channel or it's already production released?

**Comment by PlayableApps** [comment]
@liuyl how do we use it in our app ?

**Comment by Cliffordwh** [comment]
When will it come onto production?? Still baffles me that Google can roll out software without Unit testing this stuff!

**Comment by liuyl** [comment]
It's currently in beta channel and will roll out in the following weeks, you cannot force your App use the beta Google Play Services.
At the meantime I'd suggest disable the OCR feature on devices that may crash. 

Again we apologize for the crash. We do have unit tests, but unfortunately we didn't run these tests on all devices.

**Comment by PlayableApps** [comment]
@liuyl thank you , is it a good idea to let users updates their google play services

**Comment by Yrlec** [comment]
Two questions:
1. How come the faulty version was pushed into production after the issue had been identified? Having a beta channel serves no purpose if you ignore the issues found in the beta.
2. How come you cannot push it faster to production now that a fix has been created?

**Comment by whiteLT** [comment]
Any news?

**Comment by evis94** [comment]
@liuyl any updates?

**Comment by PlayableApps** [comment]
we are losing so many users oh god could you please hurry up this is really destroying our app

**Comment by lodenrogue** [comment]
Obviously Google doesn't care about it's API user base. If they did they would have put out an emergency fix on day 1. I've lost all faith in Google.

**Comment by duyp** [comment]
All I can do right now is disable OCR feature on listed devices as @Yrlec, thank you, and also alert all users that this feature might cause terrible crashes :(

This issue got me lots of 1 star, so sad! Hope Google team is going to fix soon.

**Comment by duyp** [comment]
@Yrlec: My phone is Galaxy J7 Prime, OCR lib still works well but your code `shouldDisableTextCheck` return true

**Comment by raulsb** [comment]
@liuyl this is the only free and efficient service that I have found for my application and the fact that it was a Google own library gave my application a little extra, my app works with the Text Recognizer as main feature, my boss still asks for a release date for the new Google Play Services, please I'm only asking for an estimated date, could you please give us an estimated date?

**Comment by liuyl** [comment]
@raulsb The release is rolling out slowly, some devices already get the fix in the past weeks.
The best estimation I can give is we plan to push to 100% by end of the year.

**Comment by evis94** [comment]
@liuyl 

> The release is rolling out slowly, some devices already get the fix in the past weeks.

Can you help understand what this means exactly? Once a release of the library with fix is ready we don't need to update the dependency on our application and create a release?

**Comment by liuyl** [comment]
@evis94 You don't need to do anything once the fix is released to 100%.

**Comment by nativ18** [comment]
This issue is also happening on Google Pixel smartphone with Play Services version 11.7.46

**Comment by nhattv18** [comment]
@liuyl Can you give me the list of devices that have been updated Google Play Services. I can unblock them on the Google Play Store.
Thanks!

**Comment by vlad2711** [comment]
I also have seem problem(device: Xiaomi mi max) my client want to implement this library, but I can't  create bad app. You fix this bug so slow.

**Comment by nhattv18** [comment]
@zebpaypeswani My device is samsung s8 (dreamlte), but currently there is no update . The latest version is 11.7.46

**Comment by zebpaypeswani** [comment]
@nhattv18 sorry it was the list of crashing devices.

**Comment by nhattv18** [comment]
My device (Samsung S8 - dreamlte) has been updated to the latest version (11.9.51) and Text API works fine. Thank development team!

**Comment by vlad2711** [comment]
In my device bug already fixed! Thanks

**Comment by PlayableApps** [comment]
i'm not getting crashes report anymore, Thank you dev team!

**Comment by radheshyamsingh24** [comment]
It's working for me also. Thanks development team!

**Comment by LucianIacob** [comment]
It's working very well for me also!!

**Comment by raulsb** [comment]
It's already working, Thanks @liuyl and development team!

**Comment by wanwjing** [comment]
Anyone knowing this ?
**Frame imageFrame = new Frame.Builder().setBitmap(bitmap).build();
TextRecognizer textRecognizer = new TextRecognizer.Builder(context).build();**

I hit this error. 
_E/TextNativeHandle: Error creating remote native handle
com.google.android.gms.dynamite.DynamiteModule$zza: No acceptable module found. Local version is 0 and remote version is 0.
                                                                                   at com.google.android.gms.dynamite.DynamiteModule.zza(Unknown Source)
                                                                                   at com.google.android.gms.internal.zzbjz.zzTU(Unknown Source)
                                                                                   at com.google.android.gms.internal.zzbkn.<init>(Unknown Source)
                                                                                   at com.google.android.gms.vision.text.TextRecognizer$Builder.build(Unknown Source)_

**Comment by liuyl** [comment]
Thanks everyone for your patience, I am glad that the bug is fixed.

@wanwjing, the error message probably means you don't have Google Play Service on your device.

**Comment by duyp** [comment]
In my application, text detection is fixed absolutely on latest version, but I got new crash on Barcode detection:

01-07 18:15:39.521 2315-2315/? A/DEBUG: *** *** *** *** *** *** *** *** *** *** *** *** *** *** *** ***
01-07 18:15:39.521 2315-2315/? A/DEBUG: Build fingerprint: 'samsung/j7xeltexx/j7xelte:6.0.1/MMB29K/J710FXXU1APDB:user/release-keys'
01-07 18:15:39.521 2315-2315/? A/DEBUG: Revision: '6'
01-07 18:15:39.521 2315-2315/? A/DEBUG: ABI: 'arm'
01-07 18:15:39.521 2315-2315/? A/DEBUG: pid: 32706, tid: 432, name: Thread-26201  >>> myPackageName <<<
01-07 18:15:39.521 2315-2315/? A/DEBUG: signal 11 (SIGSEGV), code 1 (SEGV_MAPERR), fault addr 0xc
01-07 18:15:39.561 2315-2315/? A/DEBUG:     r0 c67c2190  r1 00000042  r2 c67c2168  r3 00000000
01-07 18:15:39.561 2315-2315/? A/DEBUG:     r4 00000000  r5 c67c2168  r6 cf9eb220  r7 cf9eb334
01-07 18:15:39.561 2315-2315/? A/DEBUG:     r8 00000408  r9 00000001  sl 43b00000  fp 42840000
01-07 18:15:39.561 2315-2315/? A/DEBUG:     ip f3383060  sp cf9eb210  lr 00000018  pc d05d2c12  cpsr 000b0030
01-07 18:15:39.561 2315-2315/? A/DEBUG: backtrace:
01-07 18:15:39.561 2315-2315/? A/DEBUG:     #00 pc 00011c12  /data/data/com.google.android.gms/app_vision/barcode/libs/armeabi-v7a/libbarhopper.so
01-07 18:15:39.561 2315-2315/? A/DEBUG:     #01 pc 0000a345  /data/data/com.google.android.gms/app_vision/barcode/libs/armeabi-v7a/libbarhopper.so
01-07 18:15:39.561 2315-2315/? A/DEBUG:     #02 pc 0000331d  /data/data/com.google.android.gms/app_vision/barcode/libs/armeabi-v7a/libbarhopper.so
01-07 18:15:39.561 2315-2315/? A/DEBUG:     #03 pc 000043df  /data/data/com.google.android.gms/app_vision/barcode/libs/armeabi-v7a/libbarhopper.so
01-07 18:15:39.561 2315-2315/? A/DEBUG:     #04 pc 00001e63  /data/data/com.google.android.gms/app_vision/barcode/libs/armeabi-v7a/libbarhopper.so
01-07 18:15:39.561 2315-2315/? A/DEBUG:     #05 pc 00002a91  /data/data/com.google.android.gms/app_vision/barcode/libs/armeabi-v7a/libbarhopper.so (Java_com_google_android_gms_vision_barcode_internal_NativeBarcodeDetector_recognizeNative+40)
01-07 18:15:39.561 2315-2315/? A/DEBUG:     #06 pc 0063e70f  /data/data/com.google.android.gms/app_chimera/m/0000002a/oat/arm/DynamiteModulesA_GmsCore_prodmnc_xhdpi_release.odex (offset 0x3a5000)

App's google play service version: 11.8.0
Device's google play service version: 11.9.51

Please have a look.

**Comment by liuyl** [comment]
Hi @duyp, sorry to hear that. Could you please open a new issue for that?

**Comment by audoindechanterac** [comment]
Hi @liuyl !!
I still got this issue. I have it when I try to create TextRecognizer on my own project, it is fine when I'm running your sample. Here are my logs: 

I/Vision: Loading library libocr.so
I/Vision: libocr.so library load status: true
I/Vision: Reading ocr models from /data/user/0/com.google.android.gms/app_vision/ocr/data/models
I/native: jni_helper.cc:166 Loading model config /data/user/0/com.google.android.gms/app_vision/ocr/data/models/semanticlift_engine_0.2.bincfg
E/native: jni_helper.cc:170 GetContents failed: /data/user/0/com.google.android.gms/app_vision/ocr/data/models/semanticlift_engine_0.2.bincfg
I/native: jni_helper.cc:166 Loading model config /data/user/0/com.google.android.gms/app_vision/ocr/data/models/tfmini_lstm_engine_latin_0.2.bincfg
I/native: timer.cc:70 PhotoOcrEngine::Init (detector): 0.183 ms (elapsed)
I/native: timer.cc:70 BeamSearch::Init (mutator): 0.016 ms (elapsed)
I/native: timer.cc:70 PhotoOcrEngine::Init (recognizer): 1.087 ms (elapsed)
I/native: timer.cc:70 void ocr::photo::PhotoOcrEngine::Init(): 1.536 ms (elapsed)

My test device is a Lenovo P2a42 android 7.0.

I guess if it is working with your sample it should work on my project to. Is there any reason why this error occured ? I mean other than the one pointed out in november ?

**Comment by loihd** [comment]
Hello!
I'm still getting the issue on Asus ZenFone 5 (A501CG)
`E/native: jni_helper.cc:170 GetContents failed: /data/data/com.google.android.gms/app_vision/ocr/data/models/rpn_lstm_engine_tfmini.bincfg
E/native: jni_helper.cc:170 GetContents failed: /data/data/com.google.android.gms/app_vision/ocr/data/models/semanticlift_engine_0.2.bincfg`

Can you please check and fix the issue? Thank you
